share/completions/rsync.fish

```diff
@@ -52,19 +79,27 @@ complete -c rsync -l del -d "An alias for --delete-during"
 complete -c rsync -l delete -d "Delete files that don’t exist on sender"
 complete -c rsync -l delete-before -d "Receiver deletes before transfer (default)"
 complete -c rsync -l delete-during -d "Receiver deletes during xfer, not before"
+complete -c rsync -l delete-delay -d "Find deletions during, delete after"
 complete -c rsync -l delete-after -d "Receiver deletes after transfer, not before"
 complete -c rsync -l delete-excluded -d "Also delete excluded files on receiver"
+complete -c rsync -l ignore-missing-args -d "Ignore missing source args without error"
+complete -c rsync -l delete-missing-args -d "Delete missing source args from destination"
 complete -c rsync -l ignore-errors -d "Delete even if there are I/O errors"
 complete -c rsync -l force -d "Force deletion of dirs even if not empty"
 complete -c rsync -l max-delete -xa '(seq 0 10)' -d "Don’t delete more than NUM files"
 complete -c rsync -l max-size -xa '(seq 0 10){K,M,G}' -d "Don’t transfer any file larger than SIZE"
 complete -c rsync -l min-size -xa '(seq 0 10){K,M,G}' -d "Don’t transfer any file smaller than SIZE"
+complete -c rsync -l max-alloc -xa '(seq 0 10){K,M,G}' -d "Change process memory allocation limit"
 complete -c rsync -l partial -d "Keep partially transferred files"
 complete -c rsync -l partial-dir -xa '(__fish_complete_directories)' -d "Put a partially transferred file into DIR"
 complete -c rsync -l delay-updates -d "Put all updated files into place at end"
 complete -c rsync -s m -l prune-empty-dirs -d "Prune empty directory chains from file-list"
 complete -c rsync -l numeric-ids -d "Don’t map uid/gid values by user/group name"
+complete -c rsync -l usermap -xa '(__fish_complete_users)' -d "Custom username mapping"
+complete -c rsync -l groupmap -xa '(__fish_complete_groups)' -d "Custom username mapping"
+complete -c rsync -l chown -xa '(__fish_complete_users;__fish_complete_groups)' -d "Combined username/groupname mapping"
```

Comment:
A space after the `;` would be nice.

(also this does user:group, so maybe `(__fish_complete_users):(__fish_complete_groups)` might be helpful, or more complicated logic to only complete the group *after* the user - see `__fish_complete_user_at_hosts` for something similar)

Reply:
`(__fish_complete_users):(__fish_complete_groups)` unfortunately does not work in my testing, if I type `:` and hit TAB it removes the whole option? Is there a any simple solution for the common user:group pattern?

Reply:
Aaaah, okay, both of these provide a description (after a tab character), so only the users part is actually used (because the rest is now after a tab).

Here's what the actual chown completions do:

```fish
complete -c chown -d Username -a "(__fish_print_users):"
complete -c chown -d Username -a "(string match -r -- '.*:' (commandline -ct))(__fish_complete_groups)"
```

You could probably make a helper function that

1. if you don't have a ":" in the current token (`commandline -ct`) runs `__fish_complete_users`
2. If you do, prints the current token *plus* everything in `__fish_complete_groups`

Reply:
Okay, I think it is best to merge as it is and make such a helper function in a separate PR.

Reply:
Since `:somegroup` is valid, that might need a separate entry (untested):
 `(echo; __fish_print_users):(__fish_complete_groups)`